File: server/dotnet/sdk-dotnet-example/Infrastructure/Mappers/ServiceMapper.cs
using Business.DTOs.Services;
using OnlinePayments.Sdk.Domain;

namespace Infrastructure.Mappers;

public class ServiceMapper
{
    public static GetIINDetailsRequest Map(GetIinDetailsRequestDto request)
    {
        return new()
        {
            Bin = request.Bin
        };
    }
    
    public static GetIinDetailsResponseDto Map(GetIINDetailsResponse response)
    {
        return new()
        {
            PaymentProductId = response.PaymentProductId
        };
    }
}

File: server/nodejs/src/infrastructure/mappers/service-mapper.ts
import { Domain } from 'onlinepayments-sdk-nodejs';
import { GetIinDetailsRequestDto } from '../../business/dtos/services/get-iin-details-request-dto';
import { GetIinDetailsResponseDto } from '../../business/dtos/services/get-iin-details-response-dto';

export class ServiceMapper {
    static toSdkIinDetailsRequest(dto: GetIinDetailsRequestDto): Domain.GetIINDetailsRequest {
        return {
            bin: dto.bin,
        };
    }

    static fromSdkIinDetailsResponse(response: Domain.GetIINDetailsResponse): GetIinDetailsResponseDto {
        return {
            paymentProductId: response.paymentProductId,
        };
    }
}
